api/src/main/java/jakarta/mail/Part.java

```diff
@@ -453,4 +454,17 @@ Enumeration<Header> getMatchingHeaders(String[] header_names)
      */
     Enumeration<Header> getNonMatchingHeaders(String[] header_names)
             throws MessagingException;
+
+    /**
+     * Obtains the {@link StreamProvider}.
+     * It defaults to {@link Session#getDefaultInstance(java.util.Properties, Authenticator)}.
+     *
+     * @return the StreamProvider.
+     * @throws MessagingException if errors.
+     *
+     * @since JavaMail 2.2
+     */
+    default StreamProvider getStreamProvider() throws MessagingException {
+        return Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();
```

Comment:
Needs the nested catches:
````
try {
   try {
       return Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();
    } catch (ServiceConfigurationError sce) {
        throw new IllegalStateException(sce);
    }
} catch (RuntimeException re) {
   throw new MessagingException("Unable to get " + StreamProvider.class.getName(), re);
}

Also I noticed there is a more specific exception of NoSuchProviderException.  This was originally for Provider implementations so it might be abuse to reuse the more specific exception.  Something to consider.

Reply:
NoSuchProviderException extends MessagingException, I think we don't need to catch it.

Reply:
I see, it makes sense to me to throw NoSuchProviderException instead of MessagingException. I think it is a good time to do this. I am going to implement it.

Reply:
I was thinking of keeping the signature `StreamProvider getStreamProvider() throws MessagingException` so we could continue to throw other specific exceptions in the future.  We still actually would throw a more specific exception `new NoSuchProviderException` even though the signature was more general.